vars/abapEnvironmentPipeline.groovy

```diff
@@ -0,0 +1,51 @@
+void call(parameters) {
+    pipeline {
+        agent none
+        options {
+            skipDefaultCheckout()
+        }
+        stages {
+
+            stage('Init') {
+                steps {
+                    library 'piper-lib-os-dev'
+                    abapEnvironmentPipelineStageInit script: parameters.script
+                }
+            }
+
+            stage('Prepare System') {
+                when {expression {return parameters.script.commonPipelineEnvironment.configuration.runStage?.get(env.STAGE_NAME)}}
+                steps {
+                    abapEnvironmentPipelineStagePrepareSystem script: parameters.script
+                }
+            }
+
+            stage('Clone Repositories') {
+                when {expression {return parameters.script.commonPipelineEnvironment.configuration.runStage?.get(env.STAGE_NAME)}}
+                steps {
+                    abapEnvironmentPipelineStageCloneRepositories script: parameters.script
+                }
+            }
+
+            stage('ATC') {
+                when {expression {return parameters.script.commonPipelineEnvironment.configuration.runStage?.get(env.STAGE_NAME)}}
+                steps {
+                    abapEnvironmentPipelineStageATC script: parameters.script
+                }
+            }
+        }
+        post {
+            /* https://jenkins.io/doc/book/pipeline/syntax/#post */
+            success {buildSetResult(currentBuild)}
+            aborted {buildSetResult(currentBuild, 'ABORTED')}
+            failure {buildSetResult(currentBuild, 'FAILURE')}
+            unstable {buildSetResult(currentBuild, 'UNSTABLE')}
+            unsuccessful {
+                input "Unsuccessful build: delete system?"
```

Comment:
is there any further action to this or just as a notification.

Please make sure that this does not block an executor in Jenkins. I am not sure if the current way is blocking or not.